### ISSUE TITLE
Error: ENOTEMPTY: directory not empty, rmdir '.gen/fauna/indexes/output'

### ISSUE THREAD

**Issue by dotnetspec** [Plazide/fauna-gql-upload#17]
Thanks for your module!
I have been following [Sean Connolly's tutorial](https://seanconnolly.dev/simple-graphql-with-faunadb) which uses this package.

System info:
VSCode:
```
Version: 1.54.1
Commit: f30a9b73e8ffc278e71575118b6bf568f04587c8
Date: 2021-03-04T22:38:50.094Z
Electron: 11.3.0
Chrome: 87.0.4280.141
Node.js: 12.18.3
V8: 8.7.220.31-electron.0
OS: Linux x64 5.4.0-66-generic
```
When I run:
```
npm run fauna
```
I get:
```
> rock-paper-scissors@0.0.1 fauna ~/Downloads/Dev/Github/fauna-rock-paper-scissors
> npm run build:fauna && fauna-gql

> rock-paper-scissors@0.0.1 build:fauna ~/Downloads/Dev/Github/fauna-rock-paper-scissors
> tsc --project tsconfig.fauna.json

✔️  Successfully updated schema

(node:11570) ExperimentalWarning: The fs.promises API is experimental
(node:11570) UnhandledPromiseRejectionWarning: Error: ENOTEMPTY: directory not empty, rmdir '.gen/fauna/indexes/output'
(node:11570) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:11570) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```
So indexes, functions and roles don't get uploaded.

I have tried deleting the .gen (and the node_modules) directories re-installing/generating but I get the same error.
What else can I try to get past this issue?
Thanks ...

**Comment by Plazide** [comment]
Hey, thanks for creating the issue.

Which version of `fauna-gql-upload` are you using here? That build step is actually not necessary anymore and might cause issues. Try removing the build step from the `fauna` script and see if it works.

So the script should just be
```sh
fauna-gql
```

Your typescript files will still be built automatically by `fauna-gql-upload`.

**Comment by dotnetspec** [comment]
Version of fauna-gql-upload is:
`1.9.3`
I removed the build step and the scirpt is now:
`"fauna": "fauna-gql",`
I changed .fauna.json to
```
"fnsDir": "fauna/functions",
  "rolesDir": "fauna/roles",
  "indexesDir": "fauna/indexes"
```
However on:
`npm run fauna`
I still get:
```
> rock-paper-scissors@0.0.1 fauna ~/Downloads/Dev/Github/fauna-rock-paper-scissors
> fauna-gql

✔️  Successfully updated schema

(node:10588) ExperimentalWarning: The fs.promises API is experimental
(node:10588) UnhandledPromiseRejectionWarning: Error: ENOTEMPTY: directory not empty, rmdir 'fauna/indexes/output'
(node:10588) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:10588) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```
All that's changed is the directory reference (from .fauna.json)
I also deleted node_modules and did npm install which gave me:
```
added 48 packages from 29 contributors and audited 48 packages in 2.526s
found 0 vulnerabilities
```
Have I exhausted all the possibilities?
Thanks for your help ...

**Comment by Plazide** [comment]
Okay, I think I've found the cause of your error.

This error seems to be occurring on `node 10.X`, and that's because I'm using the `recursive` option for `fs.promises.rmdir` which was added in `node 12.10.0`. This explains the `ENOTEMPTY` error since it's trying to remove a directory that still contains files.

The fix for you would then be to use node 12.10.0 or higher.

node 10 reaches its end of life next month, so I don't think it's worth fixing this particular issue. Though, it would probably be a good idea for me to add a note in the readme which mentions the minimum node version and also add the `engines` property in `package.json`.

Let me know if updating node fixes it for you.
